src/controllers/photo.ts

```
@@ -0,0 +1,121 @@
+import PhotoModel from '../models/Photo'
+import { ICategory } from './../interfaces/category'
+import { IUser } from './../interfaces/user'
+import { FastifyRequest, FastifyReply } from 'fastify'
+import cloudinary from './../config/cloudinary'
+import md5 from 'crypto-js/md5'
+
+/**
+ * TYPES
+ */
+type ExtendedFastifyRequest = FastifyRequest<{
+  Params: {
+    id: string
+  }
+  Body: {
+    _id: string
+    user: IUser['_id'],
+    url?: string,
+    isDisplayed: boolean,
+    categories: Array<ICategory['_id']>,
+    dominantColors: Array<string>,
+    fileEncoded: string,
+    fileName: string
+  }
+}>
+
+/**
+ * CONTROLLER
+ */
+const getAll = async (request: ExtendedFastifyRequest, reply: FastifyReply) => {
+  try {
+    reply.send(await PhotoModel.find({}, '-__v').populate('categories', '-__v').populate('user', '-__v').lean())
+  } catch (error) {
+    console.error('error getAll photos: ', error)
+    reply.status(500)
+    reply.send({ error: error.message })
+  }
+}
+
+const post = async (request: ExtendedFastifyRequest, reply: FastifyReply) => {
+  try {
+    const hash = md5(Date.now() + Math.random().toString(36).replace(/[^a-z]+/g, '').substr(0, 5))
+    const fileEncoded : string = request.body.fileEncoded
+    const uploadResponse = await cloudinary.uploader.upload(fileEncoded, {
+      public_id: `photography/${hash}`
+    })
+
+    if (uploadResponse.url === undefined) {
```

You can throw an error in this case as it will be catched later

---

We can see it later if you want